File: app/src/main/java/com/situ2001/cliptokindle/text/Text.java
package com.situ2001.cliptokindle.text;

import java.io.Serializable;

//link html object
public class Text implements Serializable {
    private final String text;

    public Text(String text) {
        this.text = text;
    }

    public String getText() {
        return text;
    }

    public String toHtml() {
        return "<div><a href=\"" + text + "\" target=\"_blank\"><b>" + text + "</b></a>" + "</div><br><br><br>\n";
    }
}

File: app/src/main/java/com/situ2001/cliptokindle/text/Text.kt
package com.situ2001.cliptokindle.text

import java.io.Serializable

//link html object
class Text(val text: String) : Serializable {
    fun toHtml(): String {
        return "<div><a href=\"$text\" target=\"_blank\"><b>$text</b></a></div><br><br><br>\n"
    }

    override fun equals(other: Any?): Boolean {
        if (other is Text) return other.text == text
        return super.equals(other)
    }

    override fun hashCode(): Int {
        return text.hashCode()
    }
}